packages/zql/src/query/named.ts

```diff
@@ -23,15 +23,29 @@ type NamedQueryImpl<
   TReturnQuery extends Query<S, keyof S['tables'] & string>,
 > = (tx: SchemaQuery<S>, ...arg: TArg) => TReturnQuery;
 
+export function query<S extends Schema>(s: S): SchemaQuery<S>;
 export function query<
   S extends Schema,
   TArg extends ReadonlyArray<ReadonlyJSONValue>,
   TReturnQuery extends Query<S, keyof S['tables'] & string>,
 >(
-  _s: S,
+  s: S,
   name: string,
   fn: NamedQueryImpl<S, TArg, TReturnQuery>,
-): NamedQuery<S, TArg, TReturnQuery> {
+): NamedQuery<S, TArg, TReturnQuery>;
+export function query<
```

Comment:
This looks suspect to me. Only the first one defines the actual type. So it is a type error to call query with name as undefined.